### PR TITLE
Reduce 'android-18' duplication by putting it in $ANDROID_TARGET env variable

### LINKED ISSUE #8348: Reduce duplication of the "android-18" property in the Android build
If you look at:
https://github.com/servo/servo/search?utf8=%E2%9C%93&q=android-18

You'll see that we have copies of the string "android-18" in a project properties file, a .mk file, and some Rust code. This can probably be broken out into an environment variable set in the Mach/python driver that is expanded in the appropriate ways for the .mk file and Rust code, and gets generated into the properties file by passing the correct flags to the `android project update` command.

Instructions for building to target Android are available here:
https://github.com/servo/servo/wiki/Building-for-Android

### PR DESCRIPTION
**Pull request by mitchhentges**
Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data:
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy --faster` does not report any errors
- [x] These changes fix #8348

Either:
- [X] These changes do not require tests because ~~I'm lazy~~ it's a configuration change, and asserting default values is usually tedious and not very helpful

Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process.

---

Allegedly, this broke the build [back in the day](https://github.com/servo/servo/pull/8519).
Going to see if it's a piece of cake to land now

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11278)

### CODE REVIEW

File: support/android/apk/project.properties [hunk 1]
```diff
@@ -1 +0,0 @@
-target=android-18
```
Comment (mitchhentges) [inline support/android/apk/project.properties]: I'm going a little ballsy here, and assuming that the contents of `support/android/apk` are built with the variable specified [here](https://github.com/servo/servo/pull/11278/files#diff-40f5a7cf22f94aad059b2c1795347f5eR104).

Notice that in [`build.xml`](https://github.com/servo/servo/pull/11278/files#diff-2419f69b360f1fa393abc46d9d331d01R39), `project.properties` is no longer loaded, and `target` is assumed from the environment variable `$ANDROID_TARGET`

**Comment by aneeshusa** [comment]
Queue is empty and this looks reasonable. @bors-servo try

**Comment by aneeshusa** [comment]
Looks like the same error as last time. I'll try to repro this locally tomorrow with the servo-linux-android1 VM from [saltfs](https://github.com/servo/saltfs/), which has the same configuration as the android Buildbot builder.

**Comment by mitchhentges** [comment]
Ah, figured out why it's failing:
[`openssl.sh` sets up a `_ANDROID_API` variable](https://github.com/servo/servo/blob/master/support/android/openssl.sh#L39), which is then [used later in the same script](https://github.com/servo/servo/blob/master/support/android/openssl.sh#L128).
If  `_ANDROID_API` isn't set properly (say, if it's set to a blank environment variable), then `ANDROID_SYSROOT` is broken, and the build fails.

The underlying issue is that this PR addresses the `.servobuild`, machine-local case of specifying `$ANDROID_TARGET`. However, it doesn't address the "default" case, when no `target` is explicitly specified.

Where should such default configuration go? I can put "if blank then set to default" logic in each of the ~5 places I've changed in this PR, but it's a bad solution and it would make it tedious to change the `target` default in the future.

**Comment by mitchhentges** [comment]
(I wanted to believe)

**Comment by mitchhentges** [comment]
I've now set `$ANDROID_PLATFORM` to default to `android-18` in `command_base.py`.
Additionally, I'm using `android.platform` instead of `android.target` because that was already used (`arm-linux-androideabi`)

**Comment by mitchhentges** [comment]
I don't think that the failing test is related to my changes, though I may be wrong

File: support/android/build-apk/src/main.rs [hunk 2]
```diff
@@ -90,14 +90,18 @@ fn main() {
         process::exit(1);
     }
 
+    // Retrieve the Android SDK version target env variable
+    let default_platform = "android-18".to_owned();
+    let android_platform = env::var("ANDROID_PLATFORM").unwrap_or(default_platform);
```
Comment (aneeshusa) [inline support/android/build-apk/src/main.rs]: Let's make passing the environment variable mandatory, just like the other ANDROID_* variables (see the top of main for examples).

**Comment by aneeshusa** [comment]
@mitchhentges Looks good to me aside from the one review comment. Go ahead and squash while you're at it.
The test failure is a known intermittent: #11100.

-S-awaiting-review +S-needs-code-changes +S-needs-squash

**Comment by aneeshusa** [comment]
The commit looks good; please clean up the commit message. The last two lines were fixup commit messages that are no longer needed, and we should use full sentences in the body of the message (not the title though).

**Comment by mitchhentges** [comment]
I think I got it this time :)

**Comment by aneeshusa** [comment]
Thanks! @bors-servo r+

**Pull request merged**